docs: rewrite SAML SSO guide and split into dedicated sub-pages (#7484)
elastic/docs-content: deploy-manage/users-roles/cluster-or-deployment-auth/saml.md

Reviewer: VERY NIT: it is a bit of an anti-pattern for an overview page to be a full-blown tutorial page like this. I think, ideally, we'd see this be an overview of the basic concepts and the child pages (including a core tutorial), and then link down to the tutorial

this structure is _much nicer now_, but it doesn't really follow an ideal parent-child pattern. you could argue that this is ok because the other procedures are sub- or dependent procedures, but someone might want them individually 🤷

Author: I agree! The thing is that we are not following the overview / parent page and sub-pages / tutorials and procedures pattern in this PR. That would be a much bigger change.

Created https://github.com/elastic/docs-content-internal/issues/1586 for following this up.

# SAML authentication [saml-realm]

The {{stack}} supports SAML single sign-on (SSO) into {{kib}}, using {{es}} as a backend service. This guide helps you configure SAML SSO so that users can log in to {{kib}} using your organization's identity provider (IdP).

For a detailed walk-through of how to implement SAML authentication for {{kib}} with Microsoft Entra ID as an identity provider, refer to [Set up SAML with Microsoft Entra ID](/deploy-manage/users-roles/cluster-or-deployment-auth/saml-entra.md).

> Because this feature is designed with {{kib}} in mind, most sections of this guide assume {{kib}} is used. To learn how a custom web application could use the SAML REST APIs to authenticate users to {{es}} without {{kib}}, refer to [SAML without {{kib}}](#saml-no-kibana).

Reviewer: suggest we just link out to the dedicated page now (annoying to click and then likely just want to click again)

```suggestion
Because this feature is designed with {{kib}} in mind, most sections of this guide assume {{kib}} is used. To learn how a custom web application could use the SAML REST APIs to authenticate users to {{es}} without {{kib}}, refer to [SAML without {{kib}}](/deploy-manage/users-roles/cluster-or-deployment-auth/saml-without-kibana.md).
```

Author: Yep, that's something I was concerned about... :)

* If you're using {{ece}}, then you can configure SAML [at the installation level](/deploy-manage/users-roles/cloud-enterprise-orchestrator/saml.md), and then configure [SSO](/deploy-manage/users-roles/cloud-enterprise-orchestrator/configure-sso-for-deployments.md) for deployments.
::::

## How it works [saml-how-it-works]

In SAML terminology, your identity system is the *Identity Provider* (IdP): it authenticates users and issues SAML assertions about them. The {{stack}} acts as a *Service Provider* (SP): {{kib}} initiates and coordinates the SSO flow, and {{es}} validates the SAML assertions and issues session tokens.

To enable SSO, register the {{stack}} as a known SP within your IdP, and configure {{es}} and {{kib}} to trust and communicate with your IdP. When SAML is enabled in {{kib}}, unauthenticated users are redirected to the IdP login page by default.
 

> The {{stack}} implements SAML SSO with the SAML 2.0 Web Browser SSO profile. Because the flow is browser-based, the SAML realm is not suitable for standard REST clients. If you configure SAML for {{kib}}, also add a realm for API access, such as the [native realm](/deploy-manage/users-roles/cluster-or-deployment-auth/native.md).

Reviewer: gorgeous

To configure {{es}}, you will need a standard XML-formatted SAML *metadata* document from your IdP, which defines its capabilities and features. You should be able to download or generate it from your IdP's administration interface. You can pass it to {{es}} as a URL (preferred, so {{es}} reloads it automatically when it changes) or as a local file. For more information, refer to [Create a SAML realm in {{es}}](#saml-create-realm).

Most IdPs will provide an appropriate metadata file with all the features that the {{stack}} requires. Verify that your IdP's metadata includes the following:

* An `<EntityDescriptor>` with an `entityID` that you will configure as `idp.entity_id` in the {{es}} realm
* An `<IDPSSODescriptor>` that supports the SAML 2.0 protocol (`urn:oasis:names:tc:SAML:2.0:protocol`)
* At least one `<KeyDescriptor>` configured for signing (with `use="signing"`, or `use` left unspecified)
* A `<SingleSignOnService>` with binding of HTTP-Redirect (`urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect`)

> * If you want [Single Logout](#saml-logout): a `<SingleLogoutService>` with binding of HTTP-Redirect

Reviewer: while we spell it out before, I think it is nice for the bullets to stand alone and i'd keep the urn here too

All messages from the IdP must be signed. For `<Response>` messages, the signature can be on the response itself or on individual assertions. For `<LogoutRequest>` messages, the signature must be provided as a URL parameter, as required by the `HTTP-Redirect` binding.

> ### Prerequisites for self-managed clusters

Reviewer: can we put an applies_to tag here so people are clear on if this is vanilla self-managed or ALLLL self-managed options? I know the note below specified what it doesn't apply to, but nice to be clear

### Signing requirements

The {{stack}} requires that all messages from the IdP are signed. For authentication `<Response>` messages, the signature can be applied to either the response itself, or to the individual assertions. For `<LogoutRequest>` messages, the message itself must be signed, and the signature should be provided as a URL parameter, as required by the `HTTP-Redirect` binding.

## Prerequisites

Before you set up SAML single-sign on, you must have a [SAML IdP](#saml-guide-idp) configured.
* If you want [Single Logout](#saml-logout): a `<SingleLogoutService>` with binding of HTTP-Redirect

If you're using a self-managed cluster, then perform the following additional steps:
All messages from the IdP must be signed. For `<Response>` messages, the signature can be on the response itself or on individual assertions. For `<LogoutRequest>` messages, the signature must be provided as a URL parameter, as required by the `HTTP-Redirect` binding.

* Enable TLS for HTTP.
### Prerequisites for self-managed clusters

>     If your {{es}} cluster is operating in production mode, you must configure the HTTP interface to use SSL/TLS before you can enable SAML authentication. For more information, see [Encrypt HTTP client communications for {{es}}](/deploy-manage/security/set-up-basic-security-plus-https.md#encrypt-http-communication).

Reviewer: does this only apply to prod mode

Author: yes, although it should be more than obvious, as non-production clusters don't listen for connections coming from the outside :)

anyway added to the prereqs that we have specifically for self-managed. Good catch again!

> * Enable TLS for the {{es}} HTTP interface. SAML requires HTTPS. For more information, see [Encrypt HTTP client communications for {{es}}](/deploy-manage/security/set-up-basic-security-plus-https.md#encrypt-http-communication). If you started {{es}} [with security enabled](/deploy-manage/security/self-setup.md), TLS is already enabled.

Reviewer: can we swap the CTA and caveat here so people don't click through unless they need to? 

we might loosen this up to say "in most cases", because we start clusters w/ security enabled for all stack versions supported by this docset. enabling security is not an intentional choice now so this framing might be confusing

```suggestion
* Enable TLS for the {{es}} HTTP interface. SAML requires HTTPS. [In most installation scenarios](/deploy-manage/security/self-setup.md), TLS is already enabled. For manual setup steps, refer to [Encrypt HTTP client communications for {{es}}](/deploy-manage/security/set-up-basic-security-plus-https.md#encrypt-http-communication).
```

Author: Improved, thanks a lot for this!

> ## Configuration steps

Reviewer: prefer a sentence between headings/before numbered lists

> Register the {{stack}} as a SAML service provider in your IdP. The exact steps vary by provider, but you generally need to:

Reviewer: sentences where the first clause is before a colon and the "second" clause is a list after a colon is not great for translation/some ESL readers, so we can "complete" the sentence

```suggestion
Register the {{stack}} as a SAML service provider in your IdP. The exact steps vary by provider, but you generally need to do the following:
```

> If your IdP supports SP metadata import, you can generate an SP metadata file after [configuring the {{es}} realm](#saml-create-realm) and import it into your IdP to register the {{stack}} as a Service Provider automatically. Refer to [Generate SP metadata](#saml-sp-metadata).

Reviewer: shall we frontload this so people don't follow this process if they want to generate the SP metadata?

> If your IdP requires signed outgoing SAML messages (authentication requests or logout requests), you will also need to provide your {{es}} signing certificate to the IdP at this stage. Refer to [Signing and encryption](#saml-enc-sign) for how to generate certificates and configure signing in {{es}}.

Reviewer: if you move the tip out, you could feasibly take this out of a note and just make it a para after the core steps

or, maybe better, make it step 7: 

`7. Optional: ...`

```suggestion
If your IdP requires signed outgoing SAML messages (authentication requests or logout requests), then you also need to provide your {{es}} signing certificate to the IdP at this stage. Refer to [Signing and encryption](#saml-enc-sign) for how to generate certificates and configure signing in {{es}}.
```

* [SAML realm encryption settings](elasticsearch://reference/elasticsearch/configuration-reference/security-settings.md#ref-saml-encryption-settings)
* [SAML realm SSL settings](elasticsearch://reference/elasticsearch/configuration-reference/security-settings.md#ref-saml-ssl-settings)
:::{note}
If your IdP requires signed outgoing SAML messages (authentication requests or logout requests), you will also need to provide your {{es}} signing certificate to the IdP at this stage. Refer to [Signing and encryption](#saml-enc-sign) for how to generate certificates and configure signing in {{es}}.
:::
::::

This guide will walk you through the most common settings.
::::{step} Create a SAML realm in {{es}}
:anchor: saml-create-realm

Create a realm by adding the following to your [`elasticsearch.yml`](/deploy-manage/stack-settings.md) configuration file. Each configuration value is explained below.
Add a SAML realm to your [{{es}} configuration](/deploy-manage/stack-settings.md#configure-stack-settings) and restart the cluster for the changes to take effect.

> If you're using {{ece}} or {{ech}}, and you're using machine learning or a deployment with hot-warm architecture, you must include this configuration in the user settings section for each node type.

Reviewer: looks like we have this stale note in a couple other places!

Author: what do you mean? 

Are you talking about this?
> If you're using {{ece}} or {{ech}}, and you're using machine learning or a deployment with hot-warm architecture, you must include this configuration in the user settings section for each node type.

If so that's removed on purpose. It's not applicable anymore, as now in ECH and ECE the settings are applicable to all node types by default. Stef already removed a similar paragraph in a recent PR for another SAML related doc.

Reviewer: yeah I meant that it's in other files, e.g. oidc. agree it doesn't belong and should be removed. outside of the scope of this pr for sure

> 1. Controls realm priority. Assign SSO realms higher order values than password-based realms (native, LDAP). If you're using {{eck}}, set `order` to a value greater than the file realm (default `-100`) and native realm (default `-99`).

Reviewer: shall we specify that lower number == higher prio? it's unclear now

> Configure each setting as follows. For the full list of available settings, refer to [SAML realm settings](elasticsearch://reference/elasticsearch/configuration-reference/security-settings.md#ref-saml-settings).

Reviewer: can we contextualize these as required / core / commonly used settings so the link to the larger reference makes more sense? or, if all of the settings are here, "for more details about these settings ... "

>     * If you're using {{eck}}, then install the file as [custom configuration files](/deploy-manage/deploy/cloud-on-k8s/custom-configuration-files-plugins.md#use-a-volume-and-volume-mount-together-with-a-configmap-or-secret).

Reviewer: ```suggestion
    * If you're using {{eck}}, then install the file as a [custom configuration file](/deploy-manage/deploy/cloud-on-k8s/custom-configuration-files-plugins.md#use-a-volume-and-volume-mount-together-with-a-configmap-or-secret).
```

    :::

`idp.entity_id`
:   The identifier (SAML EntityID) that your IdP uses. Must match the `entityID` attribute in the IdP metadata exactly. The comparison is case-sensitive.

`sp.entity_id`
:   A unique identifier for your {{kib}} instance, expressed as a URI. Must match exactly the entity ID you [configure in your IdP](#saml-configure-idp). The comparison is case-sensitive. We recommend using the {{kib}} base URL.

`sp.acs`

> :   The URL of the Assertion Consumer Service within {{kib}} that receives authentication responses from your IdP, using the HTTP-POST binding. For example, `https://kibana.example.com/api/security/saml/callback`. This URL must be reachable from users' browsers — it does not need to be directly accessible by {{es}} or the IdP. If {{kib}} is behind a reverse proxy, use the public-facing URL.

Reviewer: ```suggestion
:   The URL of the Assertion Consumer Service within {{kib}} that receives authentication responses from your IdP, using the HTTP-POST binding. For example, `https://kibana.example.com/api/security/saml/callback`. This URL must be reachable from users' browsers, but does not need to be directly accessible by {{es}} or the IdP. If {{kib}} is behind a reverse proxy, use the public-facing URL.
```

`sp.logout`
:   The URL of the Single Logout service within {{kib}} that receives logout messages from your IdP. For example, `https://kibana.example.com/logout`. Required for [SAML Single Logout](#saml-logout). If not configured, {{es}} refuses all `<LogoutRequest>` messages from the IdP.

> `attributes.principal` (required)

Reviewer: why do we have `required` and `recommended` on only two settings here?

Author: yeah, it's not super clear. The reason is that they are required and recommended considering `attributes.*` settings, not the realm setting in general.

For the realm setting in general we are ignoring that.

:::

#### Attribute mapping [saml-attributes-mapping]

When a user authenticates through SAML, the IdP sends an assertion containing *attributes* (pieces of information about the user such as their username, email address, or group memberships). The `attributes.*` realm settings tell {{es}} which SAML attribute URI to use for each {{es}} user property.

:::{warning}
The attribute URIs must exactly match what your IdP sends in the assertion. A mismatch silently breaks authentication: the login might appear to succeed but the user's principal or group memberships will be missing or incorrect. Verify the exact URIs with your IdP administrator or by inspecting a captured SAML assertion.
:::

The most important mappings are `attributes.principal` and `attributes.groups`: `attributes.principal` is required and determines the username {{es}} assigns to the authenticated user, and `attributes.groups` is recommended if you want to assign roles based on IdP group memberships.

> For a complete reference of mappable user properties, special attribute names, and advanced patterns, refer to [Map SAML attributes to {{es}} user properties](/deploy-manage/users-roles/cluster-or-deployment-auth/saml-attribute-mapping.md).

Reviewer: we might also say this is a detailed guide to set this up (it carries the best practices / approaches we used to store in this doc)

::::

::::{step} Configure {{kib}} for SAML authentication
:anchor: saml-configure-kibana

Add the SAML [authentication provider](/deploy-manage/users-roles/cluster-or-deployment-auth/kibana-authentication.md) to your [{{kib}} configuration](/deploy-manage/stack-settings.md#configure-stack-settings) and restart {{kib}} for the changes to take effect.

```yaml
xpack.security.authc.providers:
  saml.saml1:
    order: 0
    realm: "saml1"  # <1>
```

1. Must match the SAML realm name defined in your {{es}} configuration.

With this configuration, {{kib}} redirects all unauthenticated users to your IdP for login.

:::{note}
If you run multiple {{kib}} instances connected to the same cluster or deployment (for example, behind a load balancer), apply this SAML provider configuration to every instance. If each instance is accessed through a different URL, each needs its own SAML realm. Refer to [Multiple {{kib}} instances and URLs](#_operating_multiple_kib_instances).
:::

#### Allow both SAML and basic authentication [saml-kibana-basic]

> To let some users (for example, local administrators) log in with a username and password, enable a basic provider alongside SAML:

Reviewer: there are _a lot_ of parentheticals in this doc. was trying to ignore them but will state once my preference to avoid them as a rule

Author: yeah... and I included some... =)

sometimes I feel a parenthetical statement, or an em dash makes sense and makes reading easier. But I agree they are not ideal and we shouldn't abuse.

But on the other hand if we remove all of them by policy some of our sentences or paragraphs could feel worse or more difficult to process.

Reviewer: yeah - I think sometimes adding a second sentence with contextualization is a little heavy, but we should try to take a stab and fall back to parentheses only when it breaks up the narrative

```yaml
xpack.security.authc.providers:
  saml.saml1:
    order: 0
    realm: "saml1"
    description: "Log in with SSO"
  basic.basic1:
    order: 1
```

> Users will see a login selector and can choose their preferred method. Users who log in with basic authentication must have credentials in a configured {{es}} realm (such as native or LDAP).

Reviewer: ```suggestion
When multiple authentication providers are confgired, users see a login selector and can choose their preferred method. Users who log in with basic authentication must have credentials in a configured {{es}} realm such as native or LDAP.
```

> You can also adjust [session timeout settings](/deploy-manage/security/kibana-session-management.md) (`xpack.security.session.idleTimeout` and `xpack.security.session.lifespan`) to match your security requirements.

Reviewer: this sentence feels a little throwaway here. is it part of the multiple auth providers case? perhaps needs to move up

::::

::::{step} Configure role mappings
:anchor: saml-role-mapping

SAML authentication identifies users to the {{stack}}, but does not automatically grant them any access. You must map SAML users to {{es}} roles before they can do anything.

> Role mappings connect SAML user identities to {{es}} roles, but the roles themselves must exist first. You can use built-in roles such as `superuser` or `kibana_admin` for initial testing. For production, create [custom roles](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles.md) with appropriate access to your data and [{{kib}} features](/deploy-manage/users-roles/cluster-or-deployment-auth/kibana-privileges.md#kibana-feature-privileges).

Reviewer: > appropriate access to your data

this is more than data right? it's also es features

Author: the key is to highlight the Kibana privileges section that is available when a creating role, I guess those are mapped to stack features data and kibana access to certain parts of the UI.

Because SAML is intended for kibana access we want to avoid a reader adding permissions to the data but forgetting granting basic Kibana access.

Anyway I'll try to rephrase a bit.

> You can create role mappings in the **Role Mappings** page in {{kib}} or with the [role mapping API]({{es-apis}}operation/operation-security-put-role-mapping). For the general concepts, UI workflow, and rule syntax, refer to [Map external users and groups to roles](/deploy-manage/users-roles/cluster-or-deployment-auth/mapping-users-groups-to-roles.md). The examples below show common SAML patterns.

Reviewer: ```suggestion
You can create role mappings in the **Role Mappings** page in {{kib}} or with the [role mapping API]({{es-apis}}operation/operation-security-put-role-mapping). For the general concepts, UI workflow, and rule syntax, refer to [Map external users and groups to roles](/deploy-manage/users-roles/cluster-or-deployment-auth/mapping-users-groups-to-roles.md). The following examples show common SAML patterns.
```

### Signing and encryption [saml-enc-sign]

Depending on your IdP, you might need to sign outgoing SAML messages (authentication or logout requests), decrypt encrypted assertions, or both. {{es}} supports both cases using X.509 certificates configured in the SAML realm.

For a complete example including certificate generation and configuration for PEM, PKCS#12, and JKS formats, refer to [Configure SAML signing and encryption](/deploy-manage/users-roles/cluster-or-deployment-auth/saml-signing-encryption.md).

For available settings, refer to [SAML realm signing settings](elasticsearch://reference/elasticsearch/configuration-reference/security-settings.md#ref-saml-signing-settings) and [SAML realm encryption settings](elasticsearch://reference/elasticsearch/configuration-reference/security-settings.md#ref-saml-encryption-settings).

### SAML Single Logout [saml-logout]

> The SAML protocol supports Single Logout (SLO), which ends both the {{kib}} session and the IdP session when a user logs out. Support for SLO varies between identity providers. Consult your IdP's documentation to determine what Logout services it offers.

Reviewer: ```suggestion
The SAML protocol supports Single Logout (SLO), which ends both the {{kib}} session and the IdP session when a user logs out. Support for SLO varies between identity providers. Consult your IdP's documentation to determine what logout services it offers.
```

The SAML SP sends a set of Authentication Context Class Reference values in the Authentication Request, describing the restrictions to impose on the IdP. The IdP attempts to satisfy those restrictions and indicates the result in its Authentication Response. If the IdP cannot grant the requested restrictions, authentication fails.

Configure the class reference values using `req_authn_context_class_ref` in your SAML realm. For example:

```yaml
xpack.security.authc.realms.saml.saml1:
  req_authn_context_class_ref:
    - "http://schemas.microsoft.com/claims/multipleauthn"  # <1>
```

1. This example uses the Entra ID URI for MFA. The exact URI values depend on your IdP. Some standard SAML 2.0 values such as `urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport` are widely supported, but IdP-specific URIs are common. Consult your IdP's documentation.

> {{es}} supports only the `exact` comparison method. The Authentication Response must include one of the specified values; otherwise, authentication fails. For more details, refer to [`req_authn_context_class_ref`](elasticsearch://reference/elasticsearch/configuration-reference/security-settings.md#ref-saml-settings).

Reviewer: nit: link lands a little far from the target. might want to contextualize the link a little more or inject an anchor into the target page

### Generate SP metadata [saml-sp-metadata]

> The SP metadata is an XML document that describes your {{stack}} as a SAML Service Provider. It contains your entity ID, ACS URL, logout URL, and any signing or encryption certificates configured in the realm. Some Identity Providers can import this file to configure the integration automatically, instead of requiring you to enter each value individually when [registering the {{stack}} in your IdP](#saml-configure-idp).

Reviewer: ```suggestion
SP metadata is an XML document that describes your {{stack}} as a SAML Service Provider. It contains your entity ID, ACS URL, logout URL, and any signing or encryption certificates configured in the realm. Some Identity Providers can import this file to configure the integration automatically, instead of requiring you to enter each value individually when [registering the {{stack}} in your IdP](#saml-configure-idp).
```

alt:

```suggestion
The SP metadata file is an XML document that describes your {{stack}} as a SAML Service Provider. It contains your entity ID, ACS URL, logout URL, and any signing or encryption certificates configured in the realm. Some Identity Providers can import this file to configure the integration automatically, instead of requiring you to enter each value individually when [registering the {{stack}} in your IdP](#saml-configure-idp).
```

`xpack.security.authc.providers.saml.<provider-name>.realm`
:   Set this to the name of the SAML realm that you have used in your [{{es}} realm configuration](/deploy-manage/users-roles/cluster-or-deployment-auth/saml.md#saml-create-realm), for instance: `saml1`

### Supporting SAML and basic authentication in {{kib}} [saml-kibana-basic]

The SAML support in {{kib}} is designed on the expectation that it will be the primary (or sole) authentication method for users of that {{kib}} instance. However, it is possible to support both SAML and Basic authentication within a single {{kib}} instance by setting `xpack.security.authc.providers` as per the example below:

```yaml
xpack.security.authc.providers:
  saml.saml1:
    order: 0
    realm: "saml1"
  basic.basic1:
    order: 1
```

If {{kib}} is configured in this way, users are presented with a choice at the Login Selector UI. They log in with SAML or they provide a username and password and rely on one of the other security realms within {{es}}. Only users who have a username and password for a configured {{es}} authentication realm can log in via {{kib}} login form.

> Alternatively, when the `basic` authentication provider is enabled, you can place a reverse proxy in front of {{kib}}, and configure it to send a basic authentication header (`Authorization: Basic ....`) for each request. If this header is present and valid, {{kib}} will not initiate the SAML authentication process.

Reviewer: do we care about this

Author: I'll take a look, I think it should be already covered but it worths checking again.
I think yes, we should care about it.

> The following is an example of having 3 different {{kib}} instances, 2 of which use the same internal IdP, and another which uses a different IdP.

Reviewer: ```suggestion
The following is an example of settings for a cluster with three different {{kib}} instances, two of which use the same internal IdP, and another which uses a different IdP.
```

  sp.logout: "<kibana-engineering-example-url>/logout"
  attributes.principal: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn"
```

:::{note}
You can also mix authentication methods across instances: some {{kib}} instances can use SAML while others use basic authentication against another realm type (for example, [Native](/deploy-manage/users-roles/cluster-or-deployment-auth/native.md) or [LDAP](/deploy-manage/users-roles/cluster-or-deployment-auth/ldap.md)).
:::

### SAML without {{kib}} [saml-no-kibana]

If you're building a custom web application that needs to authenticate users against {{es}} using SAML (without {{kib}}), you can use the SAML REST APIs to implement the full authentication flow directly. This is relevant when you have a custom portal, a microservice-based architecture, or any scenario where the browser-based SSO flow must be handled by your own application code rather than by {{kib}}.

The application acts as an authentication proxy: it drives the SP-initiated or IdP-initiated SSO flow, exchanges the SAML response for an {{es}} access token and refresh token, and uses that token as a `Bearer` credential for subsequent requests.

> For the complete implementation guide (including service account setup, SP-initiated and IdP-initiated flows, and logout handling), refer to [SAML without {{kib}}](/deploy-manage/users-roles/cluster-or-deployment-auth/saml-without-kibana.md).

Reviewer: ```suggestion
For the complete implementation guide, including service account setup, SP-initiated and IdP-initiated flows, and logout handling, refer to [SAML without {{kib}}](/deploy-manage/users-roles/cluster-or-deployment-auth/saml-without-kibana.md).
```
